### ISSUE TITLE
basePath makes imposter wrong url matching

### ISSUE THREAD

**Issue by bflorat** [outofcoffee/imposter#136]
When setting a baseUrl in a swagger 2 file, the swagger editor GUI and CLI curl commands never match URLs : 

Swagger 2 file : 
```
swagger: "2.0"
info:
  description: "foo"
  version: "1.0.0"
  title: "Bar"
basePath: "/api"
paths:
  /item/{id}:
    patch:
      consumes:
      - "application/json"
      produces:
      - "application/json"
      parameters:
      - name: "id"
        in: "path"
        required: true
        type: "number"
      responses:
        "200":
          description: "OK"
        "400":
          description: "Error"

```

Imposter config : 
```
plugin: openapi
specFile: "sample-swagger2.yaml"
path: /api
validation:
  request: true

resources:
  - method: PATCH
    path: "/item/200109"
    response:
      staticFile: static-1.json
      statusCode: 400  
      
```
Note that if I remove the 'path' prefix option, the problem remains but generated urls contains only two '/api' instead of three.

```
docker run -ti -p 8080:8080  -v $(pwd):/opt/imposter/config outofcoffee/imposter-openapi
Picked up JAVA_TOOL_OPTIONS: -XX:+UnlockExperimentalVMOptions -XX:+UseCGroupMemoryLimitForHeap
OpenJDK 64-Bit Server VM warning: Unable to open cgroup memory limit file /sys/fs/cgroup/memory/memory.limit_in_bytes (No such file or directory)
16:54:30 DEBUG i.v.c.l.LoggerFactory - Using io.vertx.core.logging.SLF4JLogDelegateFactory
16:54:30 INFO  i.g.i.Imposter - Starting mock engine 2.13.1
16:54:31 DEBUG i.g.i.c.u.ConfigUtil - Loading configuration file: /opt/imposter/config/imposter-config.yaml
16:54:33 DEBUG i.g.i.p.PluginManager - Loaded 5 plugin(s): [js-detector, store-detector, openapi, js-nashorn-embedded, store-inmem]
16:54:33 DEBUG i.g.i.p.o.OpenApiPluginImpl - Adding mock endpoint: PATCH -> /api/item/:id
16:54:33 DEBUG i.g.i.p.o.OpenApiPluginImpl - Adding specification UI at: http://localhost:8080/_spec
16:54:33 INFO  i.g.i.Imposter - Mock engine up and running on http://localhost:8080
16:54:34 DEBUG i.g.i.p.o.s.SpecificationServiceImpl - Generating combined specification from 1 inputs
16:54:42 ERROR i.g.i.s.ResourceServiceImpl - File not found: PATCH http://localhost:8080/api/api/item/200109
```

![image](https://user-images.githubusercontent.com/203516/167682567-51af0cca-e554-47cf-8e28-e2c0de209e59.png)

**Comment by outofcoffee** [comment]
Improved this behaviour in v2.13.5.

Here is the config:
```yaml
plugin: openapi
specFile: "sample-swagger2.yaml"
validation:
  request: true

resources:
  - method: PATCH
    path: "/api/item/200109"
    response:
      staticFile: static-1.json
      statusCode: 400
```

Test added: [mock/openapi/src/test/java/io/gatehill/imposter/plugin/openapi/OAS2BasePathTest.kt](https://github.com/outofcoffee/imposter/blob/release/2.x/mock/openapi/src/test/java/io/gatehill/imposter/plugin/openapi/OAS2BasePathTest.kt)

**Comment by bflorat** [comment]
Thanks. I confirm this is fixed in 2.14.0 (didn't try 2.13.5).
